r/R/dplyr.R

```diff
@@ -309,8 +344,27 @@ collect.arrow_dplyr_query <- function(x, as_data_frame = TRUE, ...) {
     # See dataset.R for Dataset and Scanner(Builder) classes
```

Comment:
Per https://dplyr.tidyverse.org/reference/compute.html, `compute() stores results in a remote temporary table.` I'm not sure that's appropriate here (though I'm not sure it's not). I would think in our context that would entail something like

```
compute.arrow_dplyr_query <- function(x, name, ...) assign(name, collect.arrow_dplyr_query(x, as_data_frame = FALSE))
```

which I'm not opposed to having, but I don't think that's a replacement for this functionality.

Reply:
I made ARROW-11754 to consider this

Reply:
But also per that dplyr docs page:
>`collect()` retrieves data into a local tibble

I think repurposing `collect()` with `as_data_frame = FALSE` to force computation without actually returning a tibble/data frame is kinda sus (as the kids say). Despite the database-centric framing of `compute()` in the dplyr docs, I think `compute()` is really the appropriate verb to describe this action.

Reply:
I'm going to open a dplyr PR suggesting that the `compute()` generic should lose its required `name` argument. That `name` argument is a vestige from the old days before dbplyr was broken out as a separate package, and it does nothing AFAICT because every dplyr SQL backend package would use `dbplyr::compute.tbl_sql()` (which redefines the `name` argument) or would define its own `compute()` method.

Reply:
Oh, turns out Hadley already removed that stuff from the dplyr docs, and made the explanations more in-line with the generic behavior https://github.com/tidyverse/dplyr/commit/5d47c53d23ad20f9e78ae414ebc13aa55779b550#diff-097e83b67afe19875988f4550478dd52c467eb9becff25fe0521e5b320360c03

Reply:
https://github.com/tidyverse/dplyr/pull/5783

Reply:
I will open a separate PR in ARROW-11754 to propose this change. You can resolve this thread.